sensing/radar_static_pointcloud_filter/src/radar_static_pointcloud_filter_node/radar_static_pointcloud_filter_node.cpp

```diff
@@ -0,0 +1,165 @@
+  geometry_msgs::msg::Vector3Stamped velocity_stamped{};
+  velocity_stamped.vector = velocity;
+  geometry_msgs::msg::Vector3Stamped transformed_velocity_stamped{};
+  tf2::doTransform(velocity_stamped, transformed_velocity_stamped, *transform);
+  return velocity_stamped.vector;
+}
+
+geometry_msgs::msg::Vector3 compensateEgoVehicleTwist(
+  const radar_msgs::msg::RadarReturn & radar,
+  const geometry_msgs::msg::TwistWithCovariance & ego_vehicle_twist_with_covariance,
+  geometry_msgs::msg::TransformStamped::ConstSharedPtr transform)
+{
+  const geometry_msgs::msg::Vector3 radar_velocity = getVelocity(radar);
+  const geometry_msgs::msg::Vector3 v_r = getTransformedVelocity(radar_velocity, transform);
+
+  const auto v_e = ego_vehicle_twist_with_covariance.twist.linear;
+  return geometry_msgs::build<geometry_msgs::msg::Vector3>()
+    .x(v_r.x + v_e.x)
+    .y(v_r.y + v_e.y)
+    .z(v_r.z + v_e.z);
+}
+}  // namespace
+
+namespace radar_static_pointcloud_filter
+{
+using nav_msgs::msg::Odometry;
+using radar_msgs::msg::RadarReturn;
+using radar_msgs::msg::RadarScan;
+
+RadarStaticPointcloudFilterNode::RadarStaticPointcloudFilterNode(
+  const rclcpp::NodeOptions & node_options)
+: Node("radar_static_pointcloud_filter", node_options)
+{
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+
+  // Parameter Server
+  set_param_res_ = this->add_on_set_parameters_callback(
+    std::bind(&RadarStaticPointcloudFilterNode::onSetParam, this, _1));
+
+  // Node Parameter
+  node_param_.doppler_velocity_sd = declare_parameter<double>("doppler_velocity_sd", 2.0);
+
+  // Subscriber
+  transform_listener_ = std::make_shared<tier4_autoware_utils::TransformListener>(this);
+
+  sub_radar_.subscribe(this, "~/input/radar", rclcpp::QoS{1}.get_rmw_qos_profile());
+  sub_odometry_.subscribe(this, "~/input/odometry", rclcpp::QoS{1}.get_rmw_qos_profile());
+
+  sync_ptr_ = std::make_shared<Sync>(SyncPolicy(10), sub_radar_, sub_odometry_);
+  sync_ptr_->registerCallback(std::bind(&RadarStaticPointcloudFilterNode::onData, this, _1, _2));
+
+  // Publisher
+  pub_static_radar_ = create_publisher<RadarScan>("~/output/static_radar_scan", 1);
+  pub_dynamic_radar_ = create_publisher<RadarScan>("~/output/dynamic_radar_scan", 1);
+}
+
+rcl_interfaces::msg::SetParametersResult RadarStaticPointcloudFilterNode::onSetParam(
+  const std::vector<rclcpp::Parameter> & params)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  try {
+    auto & p = node_param_;
+    update_param(params, "doppler_velocity_sd", p.doppler_velocity_sd);
+  } catch (const rclcpp::exceptions::InvalidParameterTypeException & e) {
+    result.successful = false;
+    result.reason = e.what();
+    return result;
+  }
+  result.successful = true;
+  result.reason = "success";
+  return result;
+}
+
+void RadarStaticPointcloudFilterNode::onData(
+  const RadarScan::ConstSharedPtr radar_msg, const Odometry::ConstSharedPtr odom_msg)
+{
+  transform_ = transform_listener_->getTransform(
```

Comment:
Why did you make transform_ a member variable?
Is it because you want to use the past if you can't get it within 0.02 seconds?

Reply:
I don't mean anything by it. so fixed at  [e1f2437](https://github.com/autowarefoundation/autoware.universe/pull/2362/commits/e1f2437d7bd7339cfe1d780fd2ae6db7931bcbf6) with duration time

Reply:
Is error handling not required when a transform is not received?

Reply:
>If the parameter 0.01 have problem from whole architecture, please tell me the issue with its background.

I feel that 0.01 seconds is too short and therefore fails to get transform.
how about this? @miursh

Reply:
@yukkysaito 
As I said, I have no particular reason in it.
So what duration parameter is best for you? I'll fix to it.

Reply:
> So, what duration parameter is best for you? If you have, I'll fix to it.

I feel that twice the input period or 200msec is sufficient.
However, this is not the correct answer. :pray:

Reply:
Fixed at [6fd0875](https://github.com/autowarefoundation/autoware.universe/pull/2362/commits/6fd087527fae7bec88c7f4213cdcdad363735d91)

Reply:
Is the input period 100msec?

Reply:
> Is the input period 100msec?
Yes